src/tests/useFetchPosts.test.js

```diff
@@ -0,0 +1,24 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useFetchPosts from '../hooks/useFetchPosts';
+
+describe('useFetchPosts', () => {
```

Comment:
These tests are sending real requests, right? In integration tests, you usually don't send real requests. This is more for end-to-end tests (e.g. with Cypress).

I'd suggest mocking the requests with something like [jest-fetch-mock](https://www.npmjs.com/package/jest-fetch-mock) (not sure if this is the best option, there are other packages as well). This way the tests are much faster and don't fail if the Reddit API is down for example. 

In order to create mock responses, you can copy & paste real responses from the dev tools network tab and save them to the repository.